Add support for k8s 1.23

psy-q opened the pull request:
## Checklist

- [x] The PR has a meaningful title. It will be used to auto generate the
      changelog.
      The PR has a meaningful description that sums up the change. It will be
      linked in the changelog.
- [x] PR contains a single logical change (to build a better changelog).
- [ ] Update the documentation.
- [x] Categorize the PR by adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.
- [ ] Link this PR to related issues or PRs.

<!--
Thank you for your pull request. Please provide a description above and
review the checklist.

Contributors guide: ./CONTRIBUTING.md

Remove items that do not apply. For completed items, change [ ] to [x].
These things are not required to open a PR and can be done afterwards,
while the PR is open.
-->

simu reviewed (commented):
You'll probably at least also want to update line 38 to default to K8s 1.23 now. Also, depending on what you need, you probably want to update line 50 to a more recent kube-thanos version.

psy-q commented:
Thanks, I've incorporated that and also added an OCP version that should work. The version naming scheme seems to have changed from 4.9 to 4.10, the current one is for 4.11.

simu commented on jsonnetfile.jsonnet:
```diff
@@ -21,6 +22,7 @@ local kube_thanos_version_map = {
   '1.20': 'release-0.17',  // corresponds to v0.17.0
   '1.21': 'release-0.19',  // corresponds to v0.19.0
   '1.22': 'release-0.22',  // corresponds to v0.22.0
+  '1.23': 'v0.26.0',
```
K8s 1.23 is OpenShift 4.10, so this should be v0.23.0 (cf. https://github.com/openshift/cluster-monitoring-operator/blob/release-4.10/jsonnet/jsonnetfile.json#L58), and OpenShift 4.11 (1.24) should be v0.26.0:

```suggestion
  '1.23': 'v0.23.0',
  '1.24': 'v0.26.0',
```

psy-q commented:
It didn't seem to compile with v0.29.0 so now I'm trying to default to 0.26.0.

simu commented on jsonnetfile.jsonnet:
```diff
@@ -34,7 +37,7 @@ local thanos_mixin_version =
   else
     // Use most recent version if we didn't find an entry in the map
-    thanos_mixin_version_map['1.22'];
+    thanos_mixin_version_map['1.23'];
```
If you add 1.24 for the mixin, you probably want to default to 1.24 here

simu reviewed (approved):
LGTM overall, might make sense to add 1.24 to the thanos-mixin version map as well
